packages/orbit-components/src/Timeline/TimelineStep/index.js

```diff
@@ -0,0 +1,160 @@
+// @flow
+import React from "react";
+import styled, { css } from "styled-components";
+
+import defaultTheme from "../../defaultTheme";
+import Badge from "../../Badge";
+import Text from "../../Text";
+import Stack from "../../Stack";
+import CheckCircle from "../../icons/CheckCircle";
+import Circle from "../../icons/CircleEmpty";
+import AlertCircle from "../../icons/AlertCircle";
+import CloseCircle from "../../icons/CloseCircle";
+import { useStatuses, useStep } from "../TimelineContext";
+import useMediaQuery from "../../hooks/useMediaQuery";
+
+import type { Props } from "./index";
+
+const IconWrapperStyled = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  min-width: ${({ theme }) => theme.orbit.spaceLarge};
+  min-height: ${({ theme }) => theme.orbit.spaceLarge};
+  svg {
+    background: ${({ theme }) => theme.orbit.paletteWhite};
+  }
+`;
+
+IconWrapperStyled.defaultProps = {
+  theme: defaultTheme,
+};
+
+const StatusIcon = ({ status }) => {
+  if (status === "success") return <CheckCircle color="success" />;
+  if (status === "warning") return <AlertCircle color="warning" />;
+  if (status === "critical") return <CloseCircle color="critical" />;
+
+  return <Circle color="tertiary" size="small" />;
+};
+
+const renderStatus = (status, theme) => {
+  if (status === "success") return theme.orbit.colorTextSuccess;
+  if (status === "warning") return theme.orbit.colorTextWarning;
+  if (status === "critical") return theme.orbit.colorTextCritical;
+
+  return theme.orbit.paletteCloudNormalHover;
+};
+
+const ProgressLineStyled = styled.span`
+  ${({ desktop, theme, status, isLeft }) => css`
+    ${!desktop && `position: absolute`};
+    width: ${desktop ? "50%" : "2px"};
+    ${isLeft ? `left: 0` : `right: -20px`};
+    z-index: -1;
+    ${!desktop && `left: 11px`};
+    background: ${renderStatus(status, theme)};
+    height: ${desktop ? `2px` : `calc(100% + 20px)`};
+  `}
+`;
+
+ProgressLineStyled.defaultProps = {
+  theme: defaultTheme,
+};
+
+const TimeWrapperStyled = styled.div`
```

Comment:
Please don't use specific – explicit name for any of the text properties. There can be more usages rather than just `date/time` in the timeline step. Moreover, usage of this text property should be optional, therefore the condition is unnecessary. It will be required by design, it should show up and not be hidden just visually.

Reply:
It should be hidden visually, there is case, when it can be provided, but status will be something like waiting

Reply:
It should be hidden, in case, when the status is not success (aka label) and not current step, user will have to provide time, but it should be hidden and it's easier to handle height in that case

Reply:
It has to be, from design point of view that was used in the specification.
After discussion with @williamkolmacka we want to have it opened so it will be up to the developer or designer to fit their case the most.